### ISSUE TITLE
New check: checking sequence emptiness using len()

### ISSUE THREAD

**Issue by sandsbit** [PyCQA/pylint#3751]
### Is your feature request related to a problem? Please describe
Many python developers check if a sequence is empty by checking that its length equals to zero. But it isn't optimal, 
because we don't need to calculate the full length to check if it's empty. For example, PEP8 recommends using implicit booleanness to check this: 

> For sequences, (strings, lists, tuples), use the fact that empty sequences are false:
> 
> ```
> # Correct:
> if not seq:
> if seq:
> ```

As I understand, pylint already has a warning for `if len(...):` (len-as-condition), but there aren't any warnings for comparing 
length to zero.

### Describe the solution you'd like

I think it would be better to change len-as-condition warning, so it will produce warning for this case too.

### Additional context

If you agree with me, I'd like to make this changes by myself. I wanted to start contributing to open source for a long time.
Sorry, if I make any mistakes, English is not my primary language.

**Comment by CMeza99** [comment]
I agree that len-as-condition should be checked even when making an explicit comparison. But pylint's behavior is by design, see #2684. Currently pylint allows for such code as:
```python
while len(SEQ) > 0:
    pass
```

**Comment by Pierre-Sassoulas** [comment]
Hello, this seems sensible. Thank you for being willing to contribute to Pylint. You can add more functional tests in the `tests/functional/` directory and modify the code accordingly. I'll take the time to review a pull-request attached to this issue if you propose one :)

**Comment by CMeza99** [comment]
Would this situation already be handled by `pylint.extensions.comparetozero`?

**Comment by Pierre-Sassoulas** [comment]
I don't have time to check right now but yeah, I thought this case might be already handled. Maybe the thumbsup are because you want to start contributing and not because the problem affect other people, then.

You can check the existing functional tests and maybe add a test case to confirm this if there are none.

Don't worry there is a lot of issues you can contribute to :)

**Comment by sandsbit** [comment]
When I tested, I had no warnings for this case

**Comment by Pierre-Sassoulas** [comment]
Could you provide a minimal example to reproduce @sandsbit, please ?

**Comment by CMeza99** [comment]
C2001(compare-to-zero), `pylint.extensions.comparetozero`, will catch:
```python
if len("") == 0:
    pass

while len([]) == 0:
    pass
```

but will not catch greater/less than:
```python
if len("") <= 0:
    pass

while len([]) > 0:
    pass
```

**Comment by CMeza99** [comment]
none of my examples trigger `len-as-condition`.
```
❯ pylint --disable=all --enable="len-as-condition" test.py

❯ pylint --version
pylint 2.6.0-dev1
astroid 2.4.2
Python 3.8.2 (default, Feb 26 2020, 20:01:36) 
[GCC 8.3.1 20190223 (Red Hat 8.3.1-2)]
```

**Comment by sandsbit** [comment]
> Could you provide a minimal example to reproduce @sandsbit, please ?

```python
"""test.py"""
def test(a: list):
    if len(a) == 0:
        return 0
    return 1
```

When testing this I only get docstring and naming warnings.

> C2001(compare-to-zero), pylint.extensions.comparetozero, will catch:

This check will trigger any comparison to zero, so obviously it isn't a good option for searching for bad list emptiness checking.

**Comment by Pierre-Sassoulas** [comment]
Thanks for the example and future functional tests :) It looks like we definitely want to add those cases.

**Comment by jacobtylerwalls** [comment]
I almost had a patch finished for this and then looked at #2684 where there was a PR merged to do the opposite (i.e. create the current behavior). I think we should close to avoid churn.

**Comment by DanielNoord** [comment]
Let's close then!
